packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,438 @@
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
+        estimatedExpiryTime: swapDepositChannel?.estimatedExpiryAt?.valueOf(),
+        expectedDepositAmount: swapDepositChannel?.expectedDepositAmount?.toFixed(),
+        isExpired: swapDepositChannel?.isExpired,
+        openedThroughBackend: swapDepositChannel?.openedThroughBackend,
+        affiliateBrokers,
+        fillOrKillParams: swapDepositChannel?.fokMinPriceX128
+          ? {
+              retryDurationBlocks: swapDepositChannel.fokRetryDurationBlocks,
+              refundAddress: swapDepositChannel.fokRefundAddress,
+              minPrice: getPriceFromPriceX128(
+                swapDepositChannel.fokMinPriceX128.toFixed(),
+                swapDepositChannel.srcAsset,
+                swapDepositChannel.destAsset,
+              ),
+            }
+          : undefined,
+        dcaParams: swapDepositChannel?.chunkIntervalBlocks
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
+        txConfirmations: pendingDeposit?.transactionConfirmations,
+        receivedAt: swapRequest?.depositReceivedAt?.valueOf(),
+        receivedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+      },
+      swap: {
+        ...rolledSwaps,
+        totalAmountSwapped: rolledSwaps?.totalAmountSwapped.toFixed(),
+        currentChunk: rolledSwaps && getSwapFields(rolledSwaps.currentChunk),
+        ...getEgressStatusFields(
+          swapEgress,
+          swapEgress?.broadcast,
+          ignoredEgresses,
+          'SWAP',
+          swapEgressTrackerTxRef,
+          failedSwap,
+        ),
+        srcChainRequiredBlockConfirmations:
+          (internalSrcAsset && (await getRequiredBlockConfirmations(internalSrcAsset))) ??
+          undefined,
+        estimatedDurationSeconds:
+          srcAsset && destAsset && (await estimateSwapDuration({ srcAsset, destAsset })),
+        failure: failureMode
+          ? {
+              mode: failureMode,
+              reason: error,
+            }
+          : undefined,
+      },
+      refund: {
+        ...getEgressStatusFields(
+          refundEgress,
+          refundEgress?.broadcast,
+          ignoredEgresses,
+          'SWAP',
+          refundEgressTrackerTxRef,
+          failedSwap,
+        ),
+      },
+      ccm: {
+        ...ccmParams,
+        receivedBlockIndex: swapRequest?.ccmDepositReceivedBlockIndex ?? undefined,
+      },
+      boost: {
```

Comment:
i'd have these top level properties absent/null i they don't apply to a given swap/channel